src/System.Collections.NonGeneric/tests/HashtableTests.cs

```diff
@@ -1208,16 +1208,16 @@ public void SyncRoot()
             var hash1 = new Hashtable();
             var hash2 = new Hashtable();
 
-            Assert.NotEqual(hash1.SyncRoot, hash2.SyncRoot);
-            Assert.Equal(hash1.SyncRoot.GetType(), typeof(object));
+            Assert.NotSame(hash1.SyncRoot, hash2.SyncRoot);
+            Assert.Equal(hash1.SyncRoot.GetType(), typeof(Hashtable));
```

Comment:
```suggestion
            Assert.IsType<Hashtable>(hash1.SyncRoot);
```
(I have no idea if this actually works here, so be cautions with auto-accepting the suggestion.  You may want `IsAssignableFrom` instead.)

...although this and a few similar changes should really be pulled out into a separate PR.

Reply:
...totally outside the scope of what this PR is about...